src/yang/ietf-access-control-list.yang

```diff
@@ -422,145 +457,132 @@ module ietf-access-control-list {
 
           container matches {
             description
-              "The rules in this set determine what fields will be 
-               matched upon before any action is taken on them. 
-               The rules are selected based on the feature set 
+              "The rules in this set determine what fields will be
+               matched upon before any action is taken on them.
+               The rules are selected based on the feature set
                defined by the server and the acl-type defined.";
-            
-            container l2-acl {
-              if-feature l2-acl;
-              must "derived-from(../../../../acl-type, 'acl:eth-acl')";
+
+            container eth {
+              when "derived-from(../../../../acl-type, " +
+                   "'acl:eth-acl-type')";
+              if-feature match-on-eth;
               uses packet-fields:acl-eth-header-fields;
               description
-                "Rule set for L2 ACL.";
+                "Rule set for that supports eth headers.";
             }
 
-            container ipv4-acl {
-              if-feature ipv4-acl;
-	            must "derived-from(../../../../acl-type, " +
-            	   "'acl:ipv4-acl')";
+            container ipv4 {
```

Comment:
I put ipv4 and ipv6 containers in a choice so that it would only be possible to pick one of them. Similarly for udp/tcp/icmp and I prepared one for L2 as well although this model only has ethernet.

Reply:
But what if the server can match on ipv4 and ipv6, and wants to support mixed-eth-ipv4-ipv6? Same is true for udp/tcp/icmp.

Reply:
That will still be possible to do, just in different ACEs.

You can't match IPv4 and IPv6 in the same ACE, which the choice will prevent. A packet is either IPv4 or IPv6. It's not both at the same time, so trying to match on both in the same ACE is nonsensical. However, it is still possible to have one ACE match IPv4 and the next one match IPv6 (provided acl-type allows it).

Reply:
Fair enough. Have made ipv4 and ipv6 choice containers under l3 and made tcp, udp, icmp also choices under l4. Have left eth alone for now as there is no choice there.